### PR TITLE
gc_worker: resolve locks on offline stores

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/18338
Problem Summary:
Green GC should scan locks on offline stores.

### What is changed and how it works?
What's Changed:
Only skip tombstone stores.

### Related changes
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit tes

### Release note 

- Fix the issue that green GC doesn't scan locks on offline stores which may cause data incompleteness.

### CODE REVIEW

**Comment by youjiali1995** [comment]
BTW, I think we need cherry-pick it to release-3.0/3.1 too because offline stores are skipped during destroying ranges. It's possible an offline store becomes up if someone deletes a wrong store and reverts the operation. @MyonKeminta

**Comment by youjiali1995** [comment]
@MyonKeminta PTAL

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -459,7 +459,7 @@ func (w *GCWorker) checkGCInterval(now time.Time) (bool, error) {
 }
 
 // validateGCLiftTime checks whether life time is small than min gc life time.
-func (w *GCWorker) validateGCLiftTime(lifeTime time.Duration) (time.Duration, error) {
```
Comment (MyonKeminta) [inline store/tikv/gcworker/gc_worker.go:462]: Oh

File: store/tikv/gcworker/gc_worker.go [hunk 2]
```diff
@@ -790,8 +790,11 @@ const (
 // needsGCOperationForStore checks if the store-level requests related to GC needs to be sent to the store. The store-level
 // requests includes UnsafeDestroyRange, PhysicalScanLock, etc.
 func needsGCOperationForStore(store *metapb.Store) (bool, error) {
-	engineLabel := ""
+	if store.State == metapb.StoreState_Tombstone {
```
Comment (cfzjywxk) [inline store/tikv/gcworker/gc_worker.go:793]: LGTM, better add some comments to make others aware of this condition which is easy to mistake.

Comment (youjiali1995) [reply store/tikv/gcworker/gc_worker.go:796]: added

**Pull request merged**